src/components/time-picker/time-picker.tsx

```diff
@@ -666,6 +671,17 @@ export class TimePicker {
     }
   };
 
+  private getMeridiemOrder(formatParts: Intl.DateTimeFormatPart[]): number {
+    const isRTLKind = this.locale === "ar" || this.locale === "he";
```

Comment:
@anveshmekala @jcfranco this is still setting a visual style based on language and not a direction right?. Are we sure we want to do this?

Would it be more preferable to check the direction of the element instead?

---

Separate discussion, but it seems like locale only works correctly if set on the host or document. Would we want to enhance that in the future? Maybe it should use `closest` or `closestCrossShadowBoundary`?

```
@Prop({ attribute: "lang", mutable: true }) locale: string =
    document.documentElement.lang || navigator.language || "en";
```

Reply:
> @anveshmekala @jcfranco this is still setting a visual style based on language and not a direction right?. Are we sure we want to do this?
> Would it be more preferable to check the direction of the element instead?

I think we need to do this. `Intl.DateTimeFormat#formatToParts` gives us the parts in the the correct order, but there's no notion or info on RTL/LTR from it. The meridiem part gets ordered correctly in the input by the browser based on `dir` ([example](https://codepen.io/jcfranco/pen/oNooOKd)). The time picker can't rely on this since we render each part individually. 

If you have an idea on how to improve this, we can do that in a follow-up PR.

--- 

> Separate discussion, but it seems like locale only works correctly if set on the host or document. Would we want to enhance that in the future? Maybe it should use closest or closestCrossShadowBoundary?

Yeah, I think we'd want to go this route. This came up once the time picker updated the way it determines `lang`.

Reply:
Makes sense. Thanks for confirming! :)